client/datalake/archive.py

```diff
@@ -254,12 +255,16 @@ def fetch_to_filename(self, url, filename_template=None):
 
         Returns the filename written.
         '''
-        k = self._get_key_from_url(url)
-        m = self._get_metadata_from_key(k)
+        if url.startswith('s3://'):
+            k = self._get_key_from_url(url)
+            m = self._get_metadata_from_key(k)
+        else:
+            m = self._get_metadata_from_http_url(url)
         fname = self._get_filename_from_template(filename_template, m)
         dname = os.path.dirname(fname)
         self._mkdirs(dname)
-        k.get_contents_to_filename(fname)
+        with open(fname, 'wb') as fh:
+            fh.write(self.fetch(url).read())
```

Comment:
Hm. I'm pretty sure `get_contents_to_filename` will download in chunks and save as it goes. This is important for large files, which we do often handle. I'd be fine with making the new feature (i.e., fetching from http) read the whole file then write it to disk. But I think we should leave the existing behavior for s3:// urls.

Reply:
That's a great point. I think all the plumbing is in there for fetching http with a stream, too. I'll fix this up a bit.